From 02d627c8d17f76464e6fd47d8fed091f303c7bf8 Mon Sep 17 00:00:00 2001
From: Bob Paauwe <bob.j.paauwe@intel.com>
Date: Fri, 1 Apr 2016 09:20:46 -0700
Subject: [PATCH 02/44] drm/i915: Read DPCD when connector forced on.

When a DP connector is forced on using using the video=<connector>:e
kernel command line, we bypass the calls to check if the port is
connected and also bypass the call to intel_dp_detect_dpcd();  The
result is that we don't query for the sink capabilities like the
max link bandwidth.  The sink capabilities are needed to set a
mode.

The solution is to add a call to intel_dp_detect_dpcd() in the code
path for forced on DP connectors.

Signed-off-by: Bob Paauwe <bob.j.paauwe@intel.com>
Signed-off-by: Matt Roper <matthew.d.roper@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index fbaecc0..fdea3fa 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4823,6 +4823,8 @@ bool intel_digital_port_connected(struct drm_i915_private *dev_priv,
 	intel_dp->aux.i2c_nack_count = 0;
 	intel_dp->aux.i2c_defer_count = 0;
 
+	intel_dp_detect_dpcd(intel_dp);
+
 	intel_dp_set_edid(intel_dp);
 	if (is_edp(intel_dp) || intel_connector->detect_edid)
 		status = connector_status_connected;
-- 
1.9.1

